yt_idefix/data_structures.py

```diff
@@ -320,3 +321,98 @@ def _parse_parameter_file(self):
 class PlutoVtkDataset(IdefixVtkDataset):
     _version_regexp = re.compile(r"\d+\.\d+\.?\d*[-\w+]*")
     _required_header_keyword = "PLUTO"
+
+    def __init__(
+        self,
+        filename,
+        dataset_type=None,
+        unit_system="cgs",
+        units_override=None,
+        *,
+        code_dir: str | None = None,
+        geometry: str | None = None,
+        inifile=None,
+    ):
+        # Presuming code files are along with output, otherwise header file location
+        # has to be specified.
+        if code_dir is None:
+            self.code_dir = os.path.abspath(os.path.dirname(filename))
+        else:
+            self.code_dir = os.path.abspath(code_dir)
+
+        super().__init__(
+            filename,
+            dataset_type=dataset_type,
+            unit_system=unit_system,
+            units_override=units_override,
+            geometry=self.geometry,
+            inifile=inifile,
+        )
+
+    def _parse_parameter_file(self):
+        self._parse_header_file()
+        super()._parse_parameter_file()
+        self._get_time()
+
+    def _parse_header_file(self):
+        """Pluto reads some metadata from header file 'definitions.h'."""
+        constexpr = re.compile(r"CONST_\w*")
+        header_file = os.path.join(self.code_dir, "definitions.h")
+        if os.path.isfile(header_file):
+            with open(header_file) as hf:
+                for line in hf.readlines():
+                    if "GEOMETRY" in line:
+                        self.geometry = line.split()[-1].lower()
+                    if "UNIT" in line:
+                        item = line.split()
+                        key = item[1].lower()
+                        expr = item[-1]
+                        if "CONST" in expr:
+                            expr = constexpr.sub(self._get_const_for_reg, expr)
+                        self.parameters[key] = eval(expr)
+        elif self.geometry is None:
+            raise ValueError(
+                f"Header file couldn't be found in {self.code_dir}. "
+                "The 'geometry' keyword argument must be specified. "
+                "The code units are set to be 1.0 in cgs by default."
+            )
+        else:
+            warnings.warn(
+                f"Header file couldn't be found in {self.code_dir}. "
+                "The code units are set to be 1.0 in cgs by default."
+            )
+
+    def _get_const_for_reg(self, matched):
+        """Replace matched constant string with value"""
+        key = matched.group()
+        return str(pluto_def_constants[key])
+
+    def _get_time(self):
+        """Get current time from vtk.out."""
+        log_file = os.path.join(self.directory, "vtk.out")
```

Comment:
Is `"vtk.out"` a universal name for this file ? If not, it should probably be exposed

Reply:
Yes. In Pluto documents:

> Each format has an independent output frequency and an associated log ﬁle (i.e. dbl.out, ﬂt.out, vtk.out and so forth) keeping track of the dump history.

And there's no API to change it unless modify source code.

Reply:
Excellent, thank you ! Nothing to do here then